tests/virtualization/universal/patch_guests.pm

```diff
@@ -66,6 +66,18 @@ sub run {
     }
 }
 
+sub post_run_hook () {
```

Comment:
If I understand correctly, post_run_hook is relied on to clean up the lock file for successful runs. And given that vmware and hyperv tests load different test modules, so this  post_run_hook  function is put in multiple last files, right? 
It is possible that in future new tests may be required to be added, then this function has to be moved as side impacts, which maybe easy to be overlooked for who is not familiar with this ticket or when we forget. How about adding a separate test file at the end of both vmware and hyperv jobs to do lock releasing for successful jobs, eg `unlock_guest.pm`? This file can contain only this function post_run_hook.

We can even do a similar thing for creating the guest lock, so as to decouple it from patch_guest. It will be useful when the test order changes.

HDYT?

Reply:
The issue with separate unlock_guest.pm module is that if a job fails we still need to unlock the guest. I am not sure if a special setting exists that will run unlock_guest.pm even if the test fails in the middle.

Maybe one way to make this more visible is to add a message for example

record_info "Locking guest", "Creating  lock_guest file to prevent parallel tests run on guest";
record_info "Unlocking guest_lock file", "Removing lock_guest file to make the guest available for new tests";

Reply:
My earlier suggestion is for successful job's unlock. For failure job's unlock, still need to rely on post_fail_hook of all these relevant files. To make that easier, I was considering to put the unlock logic to their common BASE class. However after checking the files, I noticed that they are not referring to some common base class of VT, but more general classes and they have different base class. So it is not that proper to put such VT specific logic to a even deeper base class chasing down their inheritance tree.

Reply:
@tbaev  Hi Teodor, to align our expectations, what's your plan on the suggestion? 

 I'd recommend to update code for this thread :)  , so that those "irrelevant" files won't need to care about the unlock logic,  eg open_vm_tools and esxi_open_vm_tools. Neither do future new tests which maybe possible to be required and added. It is easier in code maintenance.

But if you strongly prefer current way, it is acceptable if some comment can be added to explain the overall solution and lock and unlock logic here, :)  .

Reply:
@alice-suse Hello Alice, I have crated more detailed comments. 

I think it is best to keep the locking and unlocking in all test modules, even if that is a little harder to maintain. 

If a new test module is created and unlocking logic is not change by the creator of the new test module, the only "issue" will be that a second test can be started during the time the new module is executed.